pkg/fanotify/fanotify.go

```diff
@@ -22,13 +23,15 @@ import (
 )
 
 func StartFanotifier(client *conn.Client, persistFile string) error {
+	persistJSONFile := fmt.Sprintf("%s.json", persistFile)
 	f, err := os.OpenFile(persistFile, os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		return errors.Wrapf(err, "failed to open persist file %q", persistFile)
 	}
-	var existedFiles = make(map[string]struct{}, 1)
+	var eventInfoList = []*conn.EventInfo{}
+	var existedFiles = make(map[string]struct{}, 10)
 	for {
-		path, err := client.GetPath()
+		eventInfo, err := client.GetEventInfo()
```

Comment:
Just name it to  GetEvent?

Reply:
Personally, I prefer `GetEventInfo` because it is not just the `fanotify` event. 😄